modules/kvformat/tests/test_kv_scanner.c

```diff
@@ -82,105 +96,98 @@ _compare_key_value(KVScanner *scanner, va_list args, gboolean *expect_more)
 
   _assert_current_key_is(scanner, kv);
 
-  kv = va_arg(args, const gchar *);
+  kv = va_get(args);
   g_assert(kv);
   _assert_current_value_is(scanner, kv);
 
   return TRUE;
 }
 
 static gboolean
-_value_was_quoted(KVScanner *scanner, va_list args, gboolean *expect_more)
+_value_was_quoted(KVScanner *scanner, VAIterator *args, gboolean *expect_more)
 {
   if (!_compare_key_value(scanner, args, expect_more))
     return FALSE;
 
-  gboolean was_quoted = va_arg(args, gboolean);
-  g_assert((was_quoted == FALSE) ||(was_quoted == TRUE));
+  gboolean was_quoted = GPOINTER_TO_SIZE(va_get(args));
+  g_assert((was_quoted == FALSE) || (was_quoted == TRUE));
```

Comment:
Thanks for the review.

This check was introduced by me in the previous recent refactor and was only present in this intermediate refactor step. Thankfully, it is not present in the final version after the last commit, because the struct array wrapper protects against such mistakes.

This check was preferred because in case of varargs, there was a small chance that invalid usage could cause misalignment/mismatching of the arguments. This could catch some of such cases, because when accidentally casting a string `pointer` to a `gsize`, you rarely get 0 or 1.